samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor

```diff
@@ -175,93 +128,157 @@
     [Parameter]
     public string Page { get; set; } = "1";
 
-    PagedResult<ListNode> NodeList = new PagedResult<ListNode>();
-    PagedResult<ListNode> pagedNodeList = new PagedResult<ListNode>();
+    public PagedResult<ListNode> NodeList = new PagedResult<ListNode>();
+    public PagedResult<ListNode> PagedNodeList = new PagedResult<ListNode>();
     public bool IsOpened { get; set; } = false;
-    public ListNode nodeData { get; set; }
-
-    private enum ActionType { Nothing, Read, Write, Call, Publish };
-    private ActionType TypeOfAction { get; set; } = ActionType.Nothing;
-    private string Response { get; set; } = string.Empty;
-    private string Value { get; set; } = string.Empty;
+    public ListNode NodeData { get; set; }
+    
 
-
-    public void PagerPageChanged(int page)
+    /// <summary>
+    /// PagerPageChanged
+    /// </summary>
+    /// <param name="page"></param>
+    /// <returns></returns>
+    public async void PagerPageChanged(int page)
     {
-        pagedNodeList = NodeList.GetPaged(page, 10);
+        PagedNodeList = NodeList.GetPaged(page, 10);
+        foreach (var node in PagedNodeList.Results)
+        {
+            // fetch the actual value
+            if (node.NodeClass == NodeClass.Variable)
+            {
+                node.Value = await BrowseManager.ReadValueAsync(endpointId, node.Id);
+            }
+        }
         NavigationManager.NavigateTo("/browser/" + supervisorId + "/" + endpointId + "/" + page);
     }
 
+    /// <summary>
+    /// OnInitializedAsync
+    /// </summary>
+    /// <returns></returns>
     protected override async Task OnInitializedAsync()
     {
         await BrowseTreeAsync(string.Empty, new List<string>(), BrowseDirection.Forward);
     }
 
+    /// <summary>
+    /// GetTreeAsync
+    /// </summary>
+    /// <param name="id"></param>
+    /// <param name="parentId"></param>
+    /// <returns></returns>
     private async Task GetTreeAsync(string id, List<string> parentId)
     {
         await BrowseTreeAsync(id, parentId, BrowseDirection.Forward);
     }
 
+    /// <summary>
+    /// GetTreeBackAsync
+    /// </summary>
+    /// <param name="id"></param>
+    /// <param name="parentId"></param>
+    /// <returns></returns>
     private async Task GetTreeBackAsync(string id, List<string> parentId)
     {
         await BrowseTreeAsync(id, parentId, BrowseDirection.Backward);
         NavigationManager.NavigateTo("/browser/" + supervisorId + "/" + endpointId);
     }
 
+    /// <summary>
+    /// BrowseTreeAsync
+    /// </summary>
+    /// <param name="id"></param>
+    /// <param name="parentId"></param>
+    /// <param name="direction"></param>
+    /// <returns></returns>
```

Comment:
ALL of these non informational function comments should be filled in or at least simplified, with a brief summary only...

Reply:
add comments